public/docs/ts/latest/guide/security.jade

```diff
@@ -0,0 +1,246 @@
+
+  The offline template compiler prevents a whole class of vulnerabilities called template injection,
+  and also greatly improves application performance. Use the offline template compiler in production
+  deployments. Do not dynamically generate templates. Angular trusts template code, so generating
+  templates, in particular containing user data, circumvents Angular's built-in protections. See the
+  [Dynamic Forms Cookbook](../cookbook/dynamic-form.html) on how to dynamically construct forms in a
+  safe way.
+
+  ### Server side XSS protection
+
+  HTML constructed on the server is vulnerable to injection attacks. When generating server side
+  HTML, e.g. for the initial page load of the Angular application, make sure to use a templating
+  language that automatically escapes values to prevent XSS vulnerabilities on the server. Do not
+  generate Angular templates on the server side using a templating language, this carries a high
+  risk of introducing template injection vulnerabilities.
+
+.l-main-section
+h2#bypass-security-apis Trusting Safe Values
+:marked
+  Sometimes applications genuinely need to include executable code, display an `<iframe>` from some
+  URL, or construct potentially dangerous URLs. To prevent automatic sanitization in this situation,
+  you can tell Angular that you inspected a value, checked how it is generated, and made sure it is
+  always secure. But **be careful**! If you trust a value that can be malicious, you will likely
+  introduce a security vulnerability into your application. If in doubt, find a professional
+  security reviewer.
+
+  You can mark a value as trusted by injecting `DomSanitizationService`, and calling one of the
+  following methods.
+
+  * `bypassSecurityTrustHtml`
+  * `bypassSecurityTrustScript`
+  * `bypassSecurityTrustStyle`
+  * `bypassSecurityTrustUrl`
+  * `bypassSecurityTrustResourceUrl`
+
+  Remember, whether a value is safe depends on context, so you need to choose the right context for
+  your intended use of the value. Imagine the following template needs to bind a URL to a
+  `javascript:alert(...)` call.
+
++makeExample('security/ts/app/bypass-security.component.html', 'dangerous-url')(format=".")
+:marked
+  Normally, Angular would automatically sanitize the URL and disable the dangerous code. To prevent
+  this, we can mark the URL value as a trusted URL using the `bypassSecurityTrustUrl` call:
+
++makeExample('security/ts/app/bypass-security.component.ts', 'trust-url')(format=".")
+figure.image-display
+    img(src='/resources/images/devguide/security/bypass-security-component.png'
+        alt='A screenshot showing an alert box created from a trusted URL')
+:marked
+  If we need to convert user input into a trusted value, it can be convenient to do so in a
+  controller method. The template below allows users to enter a YouTube video ID, and load the
+  corresponding video in an `<iframe>`. `<iframe src>` is a resource URL, because an untrusted
```

Comment:
> `<iframe src>` is a resource URL, because an untrusted source can e.g. smuggle in file downloads that unsuspecting users would execute.

Weird phrase structure, not really sure of what it is trying to say.